internal/docstore/drivertest/drivertest.go

```diff
@@ -299,47 +317,117 @@ func testCodec(t *testing.T, ct CodecTester) {
 		t.Skip("no CodecTester")
 	}
 
-	type S struct {
+	check := func(in, dec interface{}, encode func(interface{}) (interface{}, error), decode func(interface{}, interface{}) error) {
+		t.Helper()
+		enc, err := encode(in)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if err := decode(enc, dec); err != nil {
+			t.Fatal(err)
+		}
+		if diff := cmp.Diff(in, dec); diff != "" {
+			t.Error(diff)
+		}
+	}
+
+	type DocstoreRoundTrip struct {
 		N  *int
 		I  int
+		U  uint
 		F  float64
+		C  complex128
 		St string
 		B  bool
 		By []byte
 		L  []int
+		A  [2]int
 		M  map[string]bool
+		P  *string
+		T  time.Time
 	}
-	// TODO(jba): add more fields: more basic types; pointers; structs; embedding.
+	// TODO(jba): add more fields: structs; embedding.
 
-	in := S{
+	s := "bar"
+	dsrt := &DocstoreRoundTrip{
 		N:  nil,
 		I:  1,
+		U:  2,
 		F:  2.5,
+		C:  complex(3.0, 4.0),
 		St: "foo",
 		B:  true,
 		L:  []int{3, 4, 5},
+		A:  [2]int{6, 7},
 		M:  map[string]bool{"a": true, "b": false},
 		By: []byte{6, 7, 8},
+		P:  &s,
 	}
 
-	check := func(encode func(interface{}) (interface{}, error), decode func(interface{}, interface{}) error) {
-		t.Helper()
-		enc, err := encode(in)
-		if err != nil {
-			t.Fatal(err)
-		}
-		var dec S
-		if err := decode(enc, &dec); err != nil {
-			t.Fatal(err)
+	check(dsrt, &DocstoreRoundTrip{}, ct.DocstoreEncode, ct.DocstoreDecode)
+
+	// All native codecs should support these types. If one doesn't, remove it from this
```

Comment:
And then another comment here explaining that now we're testing native encode -> docstore decode and vice versa.

Or consider splitting them into separate tests?

Reply:
Added comment.